### ISSUE TITLE
Timeline doesn't work with external storage

### ISSUE THREAD

**Issue by MuQ-dev** [pulsejet/memories#647]
**Describe the bug**

I indexed my entire Nextcloud installation, and while setting the folder in memories to a folder mounted by external storage does work perfectly, meaning I can see the folders and photos, it doesn't work when setting the timeline root. It gives me this error: No valid root folder found (.nomedia?). I can confirm, that there is no .nomedia file in that folder. I confirmed that it was working with a normal local folder. Doing it with a normal folder was the only way to get started from the start screen, because setting the timeline root didn't do anything then, and does nothing in settings now.

**To Reproduce**
Install Memories, index
Set timeline folder to external storage folder

**Platform:**

- OS: Unraid Docker (Alpine)
- Browser: Edge
- Memories Version: 5.1.0
- Nextcloud Version: 26.0.1
- PHP Version: 8.1.18

- Any errors in the JS console?
GET https://DOMAIN/apps/memories/api/days 500
Post https://DOMAIN/apps/memories/api/days 500

**Comment by pulsejet** [comment]
1/ What's the name of the external storage?
2/ Where is it mounted?
3/ What does the "timeline path" in the settings look like?

**Comment by pulsejet** [comment]
Can you try placing the external storage in a "normal" folder, then set the timeline path to the outer folder?

**Comment by MuQ-dev** [comment]
> 

1. Bilder_Videos-Privat
2. In the root folder of nextcloud 
3. //Bilder_Videos-Privat

I'll try putting it in a normal folder. Could be that memories gets confused when directly accessing this external share. Could also explain the double //

**Comment by pulsejet** [comment]
LMAO I accidentally edited your comment.

> I'll try putting it in a normal folder. Could be that memories gets confused when directly accessing this external share. Could also explain the double //

That's normal. You likely hit a bug but not sure what's the best way to debug this. An identical scenario seems to work perfectly well for me.

**Comment by MuQ-dev** [comment]
Yeah no, putting the share in a folder doesn't change anything

**Comment by MuQ-dev** [comment]
Yes, the same error.

**Comment by MuQ-dev** [comment]
Placing an image while still mounting the share in that same folder doesn't work, and gives the same error. Interestingly, the Timeline view in the Folders Tab works perfectly.

**Comment by pulsejet** [comment]
Okay we need to do one more experiment.

1. Set the timeline root to `Normal Folder` (now it works)
2. Now *move* the external storage mount to `Normal Folder`
3. What happens now?

**Comment by pulsejet** [comment]
Another thing to try: comment out the following line and restart the docker container

https://github.com/pulsejet/memories/blob/76dddf6a8b973421c8291b1c7b4da5bf378dd238/lib/Db/FsManager.php#L132

**Comment by MuQ-dev** [comment]
> 

Ohhh wow that is weird. It's working now.

**Comment by pulsejet** [comment]
There's a chance that changing the name of the external storage might work then. E.g. if there's a space at the start or end then it won't work ...

**Comment by MuQ-dev** [comment]
Can't think of a reason why, but the only character that could be weird would be the _ "Bilder_Videos-Privat"

**Comment by MuQ-dev** [comment]
Yeah, it seem to be the underscore _ character. Any ideas of why? Underscores are not an uncommon character, especially in file naming.

**Comment by tstdin** [comment]
For me the timeline works without any issues and I am *using an underscore* in the external storage name (`NAS_photo`).
The external storage was configured from the admin interface and the type is SMB/CIFS. My timeline path currently consists of two folders, the first one is a local folder, the second one is the external storage - `//Photos;//NAS_photo`.

Versions: Nextcloud 26.0.1, Memories 5.1.0.

**Comment by MuQ-dev** [comment]
> For me the timeline works without any issues and I am _using an underscore_ in the external storage name (`NAS_photo`). The external storage was configured from the admin interface and the type is SMB/CIFS. My timeline path currently consists of two folders, the first one is a local folder, the second one is the external storage - `//Photos;//NAS_photo`.
> 
> Versions: Nextcloud 26.0.1, Memories 5.1.0.

Well the difference is that your folder with the underscore is in another local folder, not in the root of nextcloud.

**Comment by pulsejet** [comment]
I can't really pinpoint the issue here. Underscore is treated as any other character, do it shouldn't cause any issues. The only special character is space, only if the occurs at the start of end of the folder name.

**Comment by MuQ-dev** [comment]
> I can't really pinpoint the issue here. Underscore is treated as any other character, so it shouldn't cause any issues. The only special character is space, only if the occurs at the start of end of the folder name.

I agree, it shouldn't. Yet, it works perfectly without the underscore, and not with it. Very weird...
